bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java

```diff
@@ -213,76 +258,137 @@ public void snapshotAsync(SnapshotDescription snapshot)
   }
 
   @Override
-  public Future<Void> addColumnFamilyAsync(TableName arg0, ColumnFamilyDescriptor arg1)
+  public Future<Void> addColumnFamilyAsync(TableName tableName, ColumnFamilyDescriptor columnFamily)
       throws IOException {
-    throw new UnsupportedOperationException("addColumnFamilyAsync"); 
+    String columnName = columnFamily.getNameAsString();
+    Modification.Builder modification = Modification.newBuilder().setId(columnName)
+        .setCreate(new ColumnDescriptorAdapter()
+        .adapt((HColumnDescriptor) columnFamily).build());
```

Comment:
You shouldn't have to cast to an `HColumnDescriptor`.

Reply:
It's a hbase ColumnFamilyDescriptor what would be the best way to convert it to a ColumnFamily required by the Modification

Reply:
got it.  Casting should be fine :)